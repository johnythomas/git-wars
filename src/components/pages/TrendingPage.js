import React, { Component } from "react";
import { Grid } from "semantic-ui-react";
import User from "./User";
import { getTrendingRepos } from "../../utils/api";

class TrendingPage extends Component {
  state = {
    repos: []
  };

  componentDidMount() {
    getTrendingRepos().then(repos => this.setState(() => ({ repos })));
  }

  render() {
    return (
      <Grid columns={4}>
        {this.state.repos.map(repo => (
          <Grid.Column key={repo.id} computer={3}>
            <User repo={repo} />
          </Grid.Column>
        ))}
      </Grid>
    );
  }
}

export default TrendingPage;
