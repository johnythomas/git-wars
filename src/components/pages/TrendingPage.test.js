import React from "react";
import { shallow } from "enzyme";
import axios from "axios";
import MockAdapter from "axios-mock-adapter";
import TrendingPage from "./TrendingPage";

describe("TrendingPage", () => {
  const items = [
    {
      id: 1,
      name: "mockRepo",
      owner: {
        avatar_url: "mockImage.jpg"
      },
      stargazers_count: 200,
      forks: 12
    }
  ];

  beforeEach(() => {
    const mockAdapter = new MockAdapter(axios, { delayResponse: 0 });
    mockAdapter
      .onGet(
        "https://api.github.com/search/repositories?q=stars:%3E1+language:All&sort=stars&order=desc&type=Repositories"
      )
      .reply(200, {
        items
      });
  });

  it("should update the state after fetcheing trending repos from github API", async () => {
    const wrapper = shallow(<TrendingPage />);
    await wrapper.instance().componentDidMount();
    expect(wrapper.state().repos).toEqual(items);
  });
});
