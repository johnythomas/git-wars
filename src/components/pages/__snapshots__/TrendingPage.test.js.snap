// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TrendingPage />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [],
      "columns": 4,
    },
    "ref": null,
    "rendered": Array [],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [],
        "columns": 4,
      },
      "ref": null,
      "rendered": Array [],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
